### PR TITLE
Feature/startup switches

### PR DESCRIPTION
**Pull request by ddp5730**
- Implemented the base logic for the bike states
- Set up the initial UART debug terminal for use by an operator
- Revised CMake to build directly to executable; no need for APM library.

### CODE REVIEW

**Review by agm3067 — APPROVED** [review]

File: include/APM/ApmDevice.hpp [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+ */
+
+#ifndef APM_APMDEVICE_HPP
+#define APM_APMDEVICE_HPP
+
+#include <EVT/io/pin.hpp>
+#include <EVT/io/UART.hpp>
+#include "ApmUart.hpp"
+
+namespace APM {
```
Comment (cbolles) [inline include/APM/ApmDevice.hpp:13]: Ideally the namespace would be EVT::APM

File: include/APM/ApmDevice.hpp [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+namespace IO = EVT::core::IO;
+
+enum class ApmMode {
```
Comment (cbolles) [inline include/APM/ApmDevice.hpp:17]: Probably want to have all caps for each APM class/type/etc

File: include/APM/ApmDevice.hpp [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+     * @param baud the baudrate for the UART device
+     */
+    explicit ApmDevice(IO::GPIO &accessorySwGpio, ApmUart &apmUart, IO::GPIO &chargeSwGpio,
+                       IO::GPIO &keyOnSwGpio, IO::GPIO &vicorSwGpio);
+
+    /**
+     * Returns the ApmUart object by reference
+     * @return const reference to the ApmUart object
+     */
+    [[nodiscard]] ApmUart& getApmUart() const;
```
Comment (cbolles) [inline include/APM/ApmDevice.hpp:42]: All caps on UART

**Review by cbolles — COMMENTED** [review]
Looks really good, I just have some stylistic suggestions

File: include/APM/ApmDevice.hpp (hunk 3 above)
Comment (ddp5730) [reply include/APM/ApmDevice.hpp:42]: If the class name is updated to `APMUart` then I don't think it makes sense for this method to be `getApmUART`.  It feels inconsistent.

Comment (cbolles) [reply include/APM/ApmDevice.hpp]: Yeah I would agree, that can be left as is

**Pull request merged**